### PR TITLE
Remove MacOS 10.15 build action and add MacOS 13 build action

### PR DESCRIPTION
**Pull request by geoffthemedio**
GitHub has apparently removed 10.15 as a hosted runner option https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

### CODE REVIEW

**Comment by geoffthemedio** [comment]
@o01eg is this in the SDK BUILD.md accurate: `The SDK contains the preconfigured and -compiled build and runtime dependencies for Visual Studio on Windows and the MacOS 10.12 SDK with Xcode 10.1 or later on macOS.` for MacOS? It looks like it was built on a MacOS 11 runner, which presumably has XCode 13 but CMakeLists.txt for the SDK has `CMAKE_OSX_DEPLOYMENT_TARGET 10.9` which seems likely too old...

File: BUILD.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Required Software Dependencies
 FreeOrion depends on the following software to build:
 
   * [Visual Studio] - 2022 for Windows Desktop ; Windows only
-  * [Xcode] - 10.1 or later ; MacOS only
+  * [Xcode] - 13.2 or later ; MacOS only
```
Comment (geoffthemedio) [inline BUILD.md:28]: https://github.com/freeorion/freeorion/actions/runs/4902766692/jobs/8754806597#step:7:28

says `Check for working CXX compiler: /Applications/Xcode_13.2.1.app`

Comment (o01eg) [reply BUILD.md:28]: Looks like minimum available is `11.7` https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md
Not sure how to enforce it.

**Review by o01eg — APPROVED** [review]

File: BUILD.md (hunk 1 above)
Comment (o01eg) [reply BUILD.md:28]: Looks like `xcode-select` should work: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-ios/actions/runs/2741438390/workflow

**Comment by o01eg** [comment]
Looks like `Generate binaries` step should be disabled for `macos-13` until resolved.

**Comment by geoffthemedio** [comment]
similar looking issue with the MacOS 13 runner: https://github.com/actions/runner-images/issues/7522

**Comment by geoffthemedio** [comment]
> Looks like `Generate binaries` step should be disabled for `macos-13` until resolved.

I don't know how to do that

File: .github/workflows/build-macos.yml [hunk 2]
```diff
@@ -43,6 +43,10 @@ jobs:
         uses: actions/checkout@v3
       - name: Clean up Mono
         run: sudo ./.github/workflows/mono-uninstall.sh
+      - name: Setup - Xcode
+        if: matrix.os == 'macos-11'
+        run: |
+          sudo xcode-select -s /Applications/Xcode_11.7.app
```
Comment (geoffthemedio) [inline .github/workflows/build-macos.yml]: If XCode 13.2.1 is available on OSX 11 then I'd prefer to use and require that, unless there's some reason not to...

Comment (o01eg) [reply .github/workflows/build-macos.yml]: If we don't need to test minimum possible XCode then `Setup - Xcode` step could be removed.

**Comment by Vezzra** [comment]
> CMakeLists.txt for the SDK has CMAKE_OSX_DEPLOYMENT_TARGET 10.9 which seems likely too old...

Yep, that's too old, and can probably be raised to 10.15.

**Review by Vezzra — APPROVED** [review]
These changes look ok. Raising the minimum required macOS and Xcode versions to more recent versions shouldn't be too much of a problem.

But the macOS deployment target version should stay at least at 10.15, if possible. Only supporting macOS versions up to 3 years back to run FO seems a bit restrictive IMO.

**Comment by geoffthemedio** [comment]
> > CMakeLists.txt for the SDK has CMAKE_OSX_DEPLOYMENT_TARGET 10.9 which seems likely too old...
> 
> Yep, that's too old, and can probably be raised to 10.15.

https://github.com/freeorion/freeorion-sdk/commit/ba4da24958a8a971f94a825ad654d921ef421bec

**Comment by geoffthemedio** [comment]
needs rebase
